src/catalog/src/durable/objects.rs

```diff
@@ -615,6 +616,31 @@ impl From<SystemCatalogItemId> for CatalogItemId {
     }
 }
 
+/// A newtype wrapper for [`CatalogItemId`] that is only for the "introspection source index" namespace.
+#[derive(Debug, Copy, Clone, Ord, PartialOrd, PartialEq, Eq)]
+pub struct IntrospectionSourceIndexCatalogItemId(u64);
```

Comment:
This struct name is out of control.

Reply:
Given that the compute introspection sources are also called "logs" in various parts of the system (e.g., compute rendering, `mz_sources`), we could name this `LogIndexCatalogItemId` instead? We could also replace "introspection source index" with "log index" everywhere else in this PR, to remove a lot of noise.

Reply:
I would prefer that personally, but my understanding is that "log" is the legacy name, and we've intentionally moved toward "introspection source" as a more accurate name.

Reply:
Fair enough!